### PR TITLE
Fix #6 add initial api specification

### LINKED ISSUE #6: Add initial API specification
**Is your feature request related to a problem? Please describe.**
This issue is not related to a problem, instead it is missing core functionality.

**Pitch your idea with a user story**
As a developer I can interact with core from the module so that I don't have to implement everything again in the module.

**Describe the solution you'd like**
I would like to have clear OpenAPI schema file that describes the API.

**Describe alternatives you've considered**
There are no alternatives, this application must have API.

**Additional context**
No additional context.

### PR DESCRIPTION
**Pull request by softgitron**
**Short summary of changes in this pull request:**
This pull request provides main core and module API definitions in OpenAPI format. OpenAPI format is very useful, since it enables automatic checking of incoming API calls and provides a way to generate documentation of the API automatically.

**Short summary of important/debatable decisions and other notable choices made when working on the pull request:**
OpenAPI was selected as a language for defining APIs in potku-console.

**Other relevant information:**
This is very big and important feature. API can be partly altered in the future, but it will be increasingly harder when code base grows. Because of that all discussion is especially welcome related to this PR.

### CODE REVIEW

**Comment by softgitron** [comment]
First full core API draft is now available. Please review it. [OpenAPI editor](https://editor.swagger.io/) can be used for easier visual inspection of the specification.

**Comment by softgitron** [comment]
> First full core API draft is now available. Please review it. [OpenAPI editor](https://editor.swagger.io/) can be used for easier visual inspection of the specification.

Initial API specification for service modules has been made as well.

File: src/modules ipc library/OpenAPI/openapi_service_modules.yaml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+      operationId: "deleteTrigger"
+      responses:
+        "200":
+          description: "Trigger deleted."
+          
+  /healthCheck:
+    get:
+      tags:
+      - poll queries
+      summary: "Check that module is still active. Called by core on a regular intervall."
```
Comment (EddieTheCubeHead) [inline src/modules ipc library/OpenAPI/openapi_service_modules.yaml]: interval.* (Remove second l)

File: src/modules ipc library/OpenAPI/openapi_service_modules.yaml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+      operationId: "getHealth"
+      responses:
+        "200":
+          description: "Module is still healthy."
+          
+  /statusCheck:
+    post:
+      tags:
+      - poll queries
+      summary: "Get current active user count on the service."
```
Comment (EddieTheCubeHead) [inline src/modules ipc library/OpenAPI/openapi_service_modules.yaml]: of* the service

File: src/core/OpenAPI/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,1793 @@
...
+tags:
+- name: "potku-console API"
+  description: "Main API for potku-console."
+  externalDocs:
+    description: "Project wiki"
+    url: "https://github.com/softgitron/potku-console/wiki"
+- name: "services"
+  description: "Endpoint for controlling services (game, HTTP... servers)."
+- name: "servers"
+  description: "Endpoint for controlling virtual maschines, containers and bare metal maschines that actually run services."
```
Comment (EddieTheCubeHead) [inline src/core/OpenAPI/openapi.yaml]: virtual machines* ... ... bare metal machines

**Review by EddieTheCubeHead — COMMENTED** [review]
Minor typos otherwise really solid.

**Comment by softgitron** [comment]
Second draft of the APIs is now ready. Specification should be now pretty ready for prime time, but all comments are always welcome.

There might be still some rough edges in this specification, but I think that minor faults can be smoothed out during development process.

**Review by EddieTheCubeHead — DISMISSED** [review]
LGTM

**Pull request merged**